stage #1/module #6. IO, NIO/lesson #1. IO/content.md

```diff
@@ -0,0 +1,127 @@
+
+![img_6.png](assets/img_6.png)
+
+There are two general-purpose byte-to-character "bridge" streams: [InputStreamReader](https://docs.oracle.com/javase/8/docs/api/java/io/InputStreamReader.html) and [OutputStreamWriter](https://docs.oracle.com/javase/8/docs/api/java/io/InputStreamReader.html). 
+Use them to create character streams when there are no prepackaged character stream classes that meet your needs.
+
+
+## Closing
+
+Unfortunately, Java machine doesn’t close streams automatically, therefore every IO stream should be closed manually. 
+If we forget to close the stream, the channel will remain open and then we would end up with a resource leak. 
+There are two main options how to close streams: basic in finally block and Java 7 feature try-with-resources. 
+Let’s improve our code and close streams in appropriate way:
+![img_7.png](assets/img_7.png)
+
+We should place stream closing in finally block, because if anything goes wrong in try section, stream anyway should be closed.
+
+Using the new Java 7 feature "try-with-resources" is the second(and recommended) option:
+![img_8.png](assets/img_8.png)
+
+## Exceptions
+In this section will be given short description of common exceptions for I/O:
+- IOException – basic IO exception
+- FileNotFoundException – file not found
+- EOFException – signals that an end of file or end of stream has been reached unexpectedly during input
+- UnsupportedEncodingException – encoding is not supported.
+
+Usually, all exceptions above are caught with IOException.
+
+## Java.io.File
+
+The **File** class is Java’s representation of a file or directory **pathname**. 
+Because file and directory names have different formats on different platforms, a simple string is not adequate to name them. 
+The File class contains several methods for working with the pathname, deleting and renaming files, creating new directories, 
+listing the contents of a directory, and determining several common attributes of files and directories. 
+
+- It is an abstract representation of files and directory pathnames.
+- A pathname, whether abstract or in string form can be either absolute or relative. The parent of an abstract pathname may be obtained by invoking the getParent() method of this class.
+- First of all, we should create the File class object by passing the filename or directory name to it. A file system may implement restrictions to certain operations on the actual file-system object, such as reading, writing, and executing. These restrictions are collectively known as access permissions.
+- Instances of the File class are immutable; that is, once created, the abstract pathname represented by a File object will never change.
+
+Constructors of File Class:
+
+- File(File parent, String child): Creates a new File instance from a parent abstract pathname and a child pathname string.
```

Comment:
Could you wrap it with ``? `File(File parent, String child)`.

Reply:
Wrapped